src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -1,211 +1,195 @@
-import React from 'react';
-import noop from 'lodash/noop';
-
+import React, {
+    ReactNode,
+    MouseEventHandler,
+    useState,
+    useRef,
+    useMemo,
+    useCallback,
+    useEffect,
+} from 'react';
 import {block} from '../utils/cn';
 import {PopupPlacement} from '../Popup';
 import {Button, ButtonProps} from '../Button';
 import {Icon} from '../Icon';
 import {DotsIcon} from '../icons/DotsIcon';
 import {
+    DropdownMenuSize,
+    DropdownMenuItem,
     DropdownMenuItemMixed,
     DropdownMenuItemAction,
-    DropdownMenuItem,
-    DropdownMenuSize,
 } from './types';
 import {DropdownMenuPopup} from './DropdownMenuPopup';
 import {MenuProps} from '../Menu';
-
 import './DropdownMenu.scss';
 
 const b = block('dropdown-menu');
 
-interface DropdownMenuGeneralProps<T> {
+export type DropdownMenuProps<T> = {
```

Comment:
Why types changed?

Reply:
the exported type hasn't really changed

before the changes, the exported type was split into a couple of types to represent the props and the state of the class component, now it's no longer necessary